File: typescript-javascript/algorithms/easy/salesByMatch.ts
/**
 * @author Rithie Natan Carvalhaes Prado
 * @date 2026-05-19
 * Last modified: 2026-06-13
 * @description Sales by Match - HackerRank
 * @see {@link https://www.hackerrank.com/challenges/sales-by-match/problem}
 */
//---------- functions ----------
/**
 * Finds the number of pairs of socks with matching colors.
 * @param {number} n - The number of socks in the pile.
 * @param {number[]} ar - The colors of each sock.
 * @returns {number} - The number of pairs of socks.
 */
function sockMerchant(n: number, ar: number[]): number
{
    let countPair = 0;
    
    ar.sort((a, b) => a - b);
    
    for(let i = 0, j = 1; i < n-1; i++, j++)
    {
        if(ar[i] === ar[j])
        { countPair++; i += 1; j += 1; }
    }//end for
    
    return(countPair);
}//sockMerchant()

File: java/algorithms/easy/salesByMatch.java
/**
 * @author Rithie Natan Carvalhaes Prado
 * @date 2026-06-13
 * Last modified: 2026-06-13
 * @description Sales by Match - HackerRank
 * @see {@link https://www.hackerrank.com/challenges/sales-by-match/problem}
 */
//---------- Imports ----------
import java.util.List;


//---------- solution ----------
class Result
{
    /**
     * 
     * @param {int} n
     * @param {List<Integer>} ar
     * @return {int}
     */
    public static int sockMerchant(int n, List<Integer> ar)
    {
        int countPairs = 0;

        ar.sort(null);

        for(int i = 0; i < n - 1; i++)
        {
            if(ar.get(i).equals(ar.get(i + 1)))
            { countPairs++; i++; }
        }//end for

        return(countPairs);
    }//end sockMerchant()
}//end class